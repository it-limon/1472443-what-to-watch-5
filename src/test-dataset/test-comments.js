export const testComments = [
  {
    id: 1,
    user: {
      id: 1,
      name: `User 1`
    },
    rating: 8,
    comment: `Comment 1`,
    date: `1995-12-17T03:24:00`
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `User 2`
    },
    rating: 7,
    comment: `Comment 2`,
    date: `1996-12-17T03:24:00`
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `User 3`
    },
    rating: 5,
    comment: `Comment 3`,
    date: `1997-12-17T03:24:00`
  }
];