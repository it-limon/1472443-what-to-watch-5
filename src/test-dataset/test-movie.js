export const testMovie = {
  id: 1,
  name: `El Camino: A Breaking Bad Movie`,
  posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
  previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
  backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
  backgroundColor: `#000000`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
  rating: 7.2,
  scoresCount: 36346,
  director: `Vince Gilligan`,
  starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
  runTime: 122,
  genre: `Drama`,
  released: 2019,
  isFavorite: true
};
