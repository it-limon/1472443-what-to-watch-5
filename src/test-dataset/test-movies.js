export const testMovies = [
  {
    id: 1,
    name: `El Camino: A Breaking Bad Movie`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
    rating: 7.2,
    scoresCount: 36346,
    director: `Vince Gilligan`,
    starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
    runTime: 122,
    genre: `Drama`,
    released: 2019,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Dark Knight`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, 
                  Batman must accept one of the greatest psychological and physical tests of his ability 
                  to fight injustice.`,
    rating: 9,
    scoresCount: 2271245,
    director: `Christopher Nolan`,
    starring: [`Christian Bale`, `Heath Ledger`, `Aaron Eckhart`],
    runTime: 152,
    genre: `Crime`,
    released: 2008,
    isFavorite: true
  }
];