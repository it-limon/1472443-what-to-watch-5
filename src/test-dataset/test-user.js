export const testUser = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png`
};