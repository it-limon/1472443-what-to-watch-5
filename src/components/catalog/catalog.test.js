import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Catalog from "./catalog";
import {Provider} from "react-redux";
import {testMovies} from "../../test-dataset/test-movies";

describe(`Render Catalog with 0 shown movie`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let catalogComponent = null;

  beforeEach(() => {
    store = mockStore({
      DATA: {
        movies: []
      },
      STATE: {
        currentGenre: `All genres`,
        shownMoviesCount: 0
      }
    });

    catalogComponent = renderer.create(
        <Provider store={store}>
          <Catalog />
        </Provider>
    );
  });

  it(`Should Catalog render correctly`, () => {
    expect(catalogComponent.toJSON()).toMatchSnapshot();
  });
});

describe(`Render Catalog with 1 shown movie`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let catalogComponent = null;

  beforeEach(() => {
    store = mockStore({
      DATA: {
        movies: testMovies
      },
      STATE: {
        currentGenre: `All genres`,
        shownMoviesCount: 1
      }
    });

    store.dispatch = jest.fn();

    catalogComponent = renderer.create(
        <Provider store={store}>
          <Catalog />
        </Provider>
    );
  });

  it(`Should Catalog render correctly`, () => {
    expect(catalogComponent.toJSON()).toMatchSnapshot();
  });

});

describe(`Render Catalog with 2 shown movies`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let catalogComponent = null;

  beforeEach(() => {
    store = mockStore({
      DATA: {
        movies: testMovies
      },
      STATE: {
        currentGenre: `All genres`,
        shownMoviesCount: 2
      }
    });

    catalogComponent = renderer.create(
        <Provider store={store}>
          <Catalog />
        </Provider>
    );
  });

  it(`Should Catalog render correctly`, () => {
    expect(catalogComponent.toJSON()).toMatchSnapshot();
  });
});
