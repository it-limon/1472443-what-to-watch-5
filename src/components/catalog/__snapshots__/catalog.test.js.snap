// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Catalog Render Catalog with 0 shown movie Should Catalog render correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  />
  <div
    className="catalog__more"
  />
</section>
`;

exports[`Render Catalog Render Catalog with 1 shown movie Should Catalog render correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Drama
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          muted={true}
          poster="https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg"
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
          onClick={[Function]}
        >
          El Camino: A Breaking Bad Movie
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;

exports[`Render Catalog Render Catalog with 2 shown movies Should Catalog render correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Drama
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          muted={true}
          poster="https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg"
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
          onClick={[Function]}
        >
          El Camino: A Breaking Bad Movie
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          muted={true}
          poster="https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg"
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
          onClick={[Function]}
        >
          The Dark Knight
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  />
</section>
`;
